aws_marketplace/using_algorithms/automl/AutoML_-_Train_multiple_models_in_parallel.ipynb

```diff
@@ -0,0 +1,635 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Goal: Train a model using Automl functionality! \n",
+    "\n",
+    "A popular approach to solve a machine learning problem is to try multiple approaches for training a model by running multiple algorithms on a dataset. Based on initial analysis, you can decide which algorithm to use for training and tuning the actual model. However, each algorithm can have specific feature requirements such as data must be numeric, missing values must be addressed before the training, etc. Performing algorithm specific feature engineering tasks can take time. Such a project can be shortened by running an AutoML algorithm that performs feature engineering tasks such as one-hot encoding, generalization, addressing missing values, automatically and then trains models using multiple algorithms in parallel.  \n",
+    "\n",
+    "This notebook demonstrates how to use such an AutoML algorithm fferd by [H2o](https://aws.amazon.com/marketplace/seller-profile?id=55552124-d41b-4bad-90db-72d427682225) in AWS Marketplace for machine learning.  AutoML from H2O trains one or more of following types of models in parallel:\n",
+    "1. XGBoost GBM (Gradient Boosting Machine)\n",
+    "2. GLM \n",
+    "3. default Random Forest (DRF)\n",
+    "4. Extremely Randomized Forest (XRT)\n",
+    "5. Deep Neural Nets\n",
+    "\n",
+    "Once these models have been trained, it also creates two stacked ensemble models:\n",
+    "1. An ensemble model created using all the models.\n",
+    "2. Best of family ensemble model created using models that performed best in each class/family.\n",
+    "\n",
+    "For more information on how H2O's AutoML works, see [FAQ section of H2O's documentation.](http://docs.h2o.ai/h2o/latest-stable/h2o-docs/automl.html#faq)\n",
+    "\n",
+    "\n",
+    "#### Compatibility\n",
+    "This notebook is compatible only with [H2O-3 Automl Algorithm](https://aws.amazon.com/marketplace/pp/prodview-vbm2cls5zcnky) from AWS Marketplace and an AWS Marketplace subscription is required to successfully run this notebook. \n",
+    "\n",
+    "\n",
+    "#### Usage instructions\n",
+    "You can run this notebook one cell at a time (By using Shift+Enter for running a cell).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Let us install necessary H2O library which you would use to load and inspect the model summary.\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install http://h2o-release.s3.amazonaws.com/h2o/rel-wright/10/Python/h2o-3.20.0.10-py2.py3-none-any.whl"
```

Comment:
why not just !pip install <python wheel> ?

Reply:
To prevent issues with a known issue in Jupyter where the pip on the PATH is not the same used by the currently running kernel, pip installs should do the following so that the proper version of pip is always used